lib/Providers/OpenIDConnectProvider.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace SimpleSAML\Module\authoauth2\Providers;
+
+use Firebase\JWT;
+use League\OAuth2\Client\Provider\AbstractProvider;
+use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+use League\OAuth2\Client\Provider\GenericResourceOwner;
+use League\OAuth2\Client\Tool\BearerAuthorizationTrait;
+use League\OAuth2\Client\Token\AccessToken;
+use Psr\Http\Message\ResponseInterface;
+use SimpleSAML\Logger;
+use SimpleSAML\Utils\HTTP;
+
+class OpenIDConnectProvider extends AbstractProvider
+{
+    use BearerAuthorizationTrait;
+
+    const CONFIGURATION_PATH = '/.well-known/openid-configuration';
+    const ERROR_FIELD = 'error';
+
+    /**
+     * @var string
+     */
+    protected $issuer;
+
+    /**
+     * @var array
+     */
+    private $openIdConfiguration;
+
+    /**
+     * @var string
+     */
+    private $responseError = 'error';
+
+    public function __construct(array $options = [], array $collaborators = [])
+    {
+        parent::__construct($options, $collaborators);
+
+        if (!array_key_exists('issuer', $options))
+        {
+            throw new \InvalidArgumentException(
+                'Required options not defined: issuer'
+            );
+        }
+        $this->issuer = $options['issuer'];
+    }
+
+    protected function getScopeSeparator()
+    {
+        return ' ';
+    }
+
+    protected function getDefaultScopes()
+    {
+        return 'openid profile';
```

Comment:
Same with this. If this is not ever going to change, I think it makes sense to have it as a constant in the class, so that we reduce the amount of methods and make the values available to anybody using the class, if that ever makes sense.

Reply:
Same response as above